src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,274 @@
+package main.java.gui;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.util.ArrayList;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.WindowConstants;
+import java.awt.event.*;
+
+/**
+ * GUI pop-up responsible for allowing the user to edit and add different rules
+ * for visualizing the excel table.
+ * 
+ * @author Hugo Barroca
+ */
+public class EditRulePopup {
+	private ArrayList<String> ruleMetrics = new ArrayList<String>();
+	private JFrame frame;
+	private JTextField nameText;
+	private JScrollPane metricsScrollpane;
+	private JComboBox<String> condition;
+
+	private JPanel namePanel;
+	private JPanel metricsPanel;
+	private JPanel metricsListPanel;
+	private JPanel addNewMetricPanel;
+	private JPanel controlPanel;
+
+	private final int FRAME_X = 685;
+	private final int FRAME_Y = 300;
+
+	/**
+	 * Constructs and initializes the GUI pop-up.
+	 */
+	public EditRulePopup() {
+		frame = new JFrame("Personalized Rules");
+		frame.setLayout(new BorderLayout());
+		frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);

```

Comment:
This popup should not have this. It should not terminate the application on close. It should just close the window.
```suggestion
```

Reply:
Here's an example of how to change the close window listener to something else (taken from mine and fran's PCD project last year's):

frame.addWindowListener(new java.awt.event.WindowAdapter() {
			@Override
			public void windowClosing(WindowEvent e) {
				user.unsubscribeFromDiretorio();
				e.getWindow().dispose();
				System.exit(0);
			}
		});

Reply:
I imagine we want to confirm save on exit and when closing this popup, go back to the main frame

Reply:
I think we never close the previous window when opening this one. So it shouldn't be a problem.

Reply:
About the event listener thing, do we need to have a "on close" action? It will just close the pop up, so I don't think we need to attach anything to this event.